src/worker/git.rs

```diff
@@ -17,12 +17,21 @@ pub fn add_crate(env: &Environment, krate: Crate) -> Result<(), PerformError> {
 
     // Add the crate to its relevant file
     fs::create_dir_all(dst.parent().unwrap())?;
-    let mut file = OpenOptions::new().append(true).create(true).open(&dst)?;
+    let mut file = OpenOptions::new()
+        .append(true)
+        .create(true)
+        .read(true)
+        .open(&dst)?;
     serde_json::to_writer(&mut file, &krate)?;
     file.write_all(b"\n")?;
 
-    let message: String = format!("Updating crate `{}#{}`", krate.name, krate.vers);
+    file.rewind()?;
+    let mut contents = String::new();
+    file.read_to_string(&mut contents)?;
+    env.uploader
+        .upload_index(env.http_client(), &krate.name, contents)?;
```

Comment:
Here the HTTP index is being updated before the index changes are pushed to GitHub. At least for now, I would like to keep the git index repository as the authoritative index, so we should make sure that the git push succeeds before updating the HTTP index. I think the best solution would be to enqueue another background job that copies the file from the index over the S3. On the rare occasion that S3 goes down between the original publish and this task running, then it will be retried until successful and will be visible with our existing monitoring.